### PR TITLE
[State Observability] Set the default detail formatting as yaml + quicker head node register

### LINKED ISSUE #26939: [State Observability] Fix the first 5 seconds delay until the head node is registered.
### What happened + What you expected to happen

When the API server starts, it queries the list of all nodes in the cluster every 5 seconds. This means after the cluster starts, for the first 5 seconds, the head node is not registered to the API server (which means the state API will output empty data for the first 5 seconds). We should fix this by immediately register the head node when the API server starts.

### Versions / Dependencies

Master

### Reproduction script

N/A

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR does 2 things.

1. When `--detail` is specified, set the default formatting as yaml. 
2. It seems like it takes 5 seconds to register the head node to the API server (because it gets node info every 5 second, and when the API server just starts, the head node is not registered to GCS). It decreases the node ping frequency until the head node is registered to API server. 

## Related issue number

Closes https://github.com/ray-project/ray/issues/26939

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/node/node_head.py [hunk 1]
```diff
@@ -146,7 +161,31 @@ async def _update_nodes(self):
...
-                await asyncio.sleep(node_consts.UPDATE_NODES_INTERVAL_SECONDS)
+                self._node_update_cnt += 1
+                # Until the head node is registered, we update the
+                # node status more frequently.
+                # If the head node is not updated after 10 seconds, it just stops
+                # doing frequent update to avoid unexpected edge case.
+                if (
+                    not self._head_node_registration_time_s
+                ) and self._node_update_cnt < 100:
+                    await asyncio.sleep(0.1)
```
Comment (rickyyx) [inline dashboard/modules/node/node_head.py:172]: Could we make those magical numbers into vars/consts? E.g.  the 10 seconds and the 0.1 second frequency.

File: dashboard/modules/node/node_head.py [hunk 2]
```diff
@@ -146,7 +161,31 @@ async def _update_nodes(self):
             except Exception:
                 logger.exception("Error updating nodes.")
             finally:
-                await asyncio.sleep(node_consts.UPDATE_NODES_INTERVAL_SECONDS)
+                self._node_update_cnt += 1
+                # Until the head node is registered, we update the
+                # node status more frequently.
+                # If the head node is not updated after 10 seconds, it just stops
```
Comment (rickyyx) [inline dashboard/modules/node/node_head.py:167]: Should there be some sort of logging for better visbility if head node is not updated after 10secs?

**Review by rickyyx — APPROVED** [review]
The table->yaml print looks good to me! Have a few questions w.r.t the quicker head node registration: 

> It seems like it takes 5 seconds to register the head node to the API server (because it gets node info every 5 second, and when the API server just starts, the head node is not registered to GCS). It decreases the node ping frequency until the head node is registered to API server.

1. What does `registering head node to the API server` mean? 
2. What kind of error would surface w/o this change?

**Comment by rkooo567** [comment]
@rickyyx 

> What does registering head node to the API server mean?

We queries the GCS every 5 second to obtain all existing node states. And then we use that information to create stubs from the cluster. 

The problem is the API server starts before the head node raylet is registered to GCS, meaning it will take at least 5 seconds to register the head node raylet to the API serve. 

> What kind of error would surface w/o this change?

After the cluster starts, it takes at least 5 seconds to access any state API (it will say resources are not available although they are). I think an alternative solution is to not reply any RPC until the head node is registered.

**Comment by rkooo567** [comment]
Btw, eventually we want better mechanism than this.. we can revisit this when we rewrite API server in the sooner future

**Pull request merged**